### PR TITLE
fix(15859,16095): Added divider to NumberInput,Unintended Divider/Outline removed, Avoids magic numbers 

### LINKED ISSUE #15859: Introduce Divider across multiple states for AI Number Input
image

### PR DESCRIPTION
**Pull request by 2nikhiltom**
Closes #15859 & #16095

- Added dividers to `NumberInput` across all states (warn/invalid)
- Fixes :  Unintended Divider/Outline Appears When Input is Focused or is Invalid
- Removed Magic numbers

 
**New**

- Dividers in `NumberInput`
- Styles fixes
- Magic numbers removed

**Changed**

-  Styles changes  

**Removed**

- Magic Numbers removed

#### Testing / Reviewing

1. Open story book
2. Open unstable--slug component under experimental_folder
3. Expand examples sub-folder and open numberInput component
4. Test the dividers across all states (warn/invalid)

### CODE REVIEW

**Review by guidari — APPROVED** [review]
LGTM! 🚀

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 1]
```diff
@@ -174,6 +173,7 @@
 
     &:hover {
       background-color: $field-hover;
+      box-shadow: 0 -1px 0 var(--cds-border-strong) inset;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
      box-shadow: 0 -1px 0 $border-strong inset;
```

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 2]
```diff
@@ -206,31 +206,39 @@
     input[type='number']:focus
     ~ .#{$prefix}--number__controls
     .#{$prefix}--number__control-btn {
-    border-block-end-color: $focus;
-    box-shadow: inset 0 2px $focus, inset 0 -1px $focus;
+    border-width: 2px 0;
+
+    border-style: solid;
+    border-color: var(--cds-focus, #0f62fe);
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
    border-width: 2px 0;
    border-style: solid;
    border-color: $focus;
```

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 3]
```diff
@@ -206,31 +206,39 @@
...
+    outline-offset: -2px;
   }
 
   .#{$prefix}--number
     input[type='number']:focus
     ~ .#{$prefix}--number__controls
-    .#{$prefix}--number__control-btn:last-of-type {
-    box-shadow: inset 0 2px $focus, inset 0 -1px $focus, inset -2px 0 $focus;
+    .#{$prefix}--number__control-btn:last-of-type:hover {
+    box-shadow: -4px 0 0 -2px #0f62fe inset;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
    box-shadow: -4px 0 0 -2px $focus inset;
```

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 4]
```diff
@@ -206,31 +206,39 @@
...
   }
 
   // add invalid box-shadow to number controls only when invalid input is not focused
   .#{$prefix}--number
     input[type='number'][data-invalid]:not(:focus)
     ~ .#{$prefix}--number__controls
     .#{$prefix}--number__control-btn:last-of-type:hover {
-    box-shadow: inset 0 2px $support-error, inset 0 -1px $support-error,
-      inset -2px 0 $support-error;
+    box-shadow: -4px 0 0 -2px #da1e28 inset;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
    box-shadow: -4px 0 0 -2px $support-error inset;
```

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 5]
```diff
@@ -206,31 +206,39 @@
...
 
   .#{$prefix}--number
     input[type='number'][data-invalid]:not(:focus)
     ~ .#{$prefix}--number__controls
     .#{$prefix}--number__control-btn:hover {
-    box-shadow: inset 0 2px $support-error, inset 0 -1px $support-error;
+    border-width: 2px 0;
+    border-style: solid;
+
+    border-color: var(--cds-support-error, #da1e28);
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
    border-width: 2px 0;
    border-style: solid;
    border-color: $support-error;
```

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 6]
```diff
@@ -490,3 +529,20 @@
...
+  display: none;
+}
+
+.#{$prefix}--number__input-wrapper--slug .#{$prefix}--number__rule-divider {
+  display: none;
+}
+
+.#{$prefix}--number__input-wrapper--slug
+  .#{$prefix}--number__control-btn:hover {
+  box-shadow: 0 -1px 0 var(--cds-ai-border-strong, #4589ff) inset;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/number-input/_number-input.scss]:
```suggestion
  box-shadow: 0 -1px 0 $ai-border-strong inset;
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Overall looks good, just a couple suggestions to make sure we're using the correct tokens vs. hard coded values.

**Comment by 2nikhiltom** [comment]
Thanks @alisonjoseph for looking into this, styles are now updated with correct Saas tokens

**Review by alisonjoseph — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Comment by tay1orjones** [comment]
@Kritvi-bhatia17 this one just needs a visual review

**Comment by Kritvi-bhatia17** [comment]
> @Kritvi-bhatia17 this one just needs a visual review

Hi @tay1orjones! @alina-jacob has worked on the dropdown and would be better positioned to review it since she has more context.

**Review by alina-jacob — APPROVED** [review]
Looks good to me Nikhil! 🚀 
Thanks!

**Pull request merged**